packages/devtools_app/lib/src/shared/analytics/analytics_controller.dart

```diff
@@ -43,8 +46,17 @@ class AnalyticsController {
 
   final AsyncAnalyticsCallback? onDisableAnalytics;
 
+  final AsyncAnalyticsCallback? _confirmConsentMessageShown;
+  Future<void> confirmConsentMessageShown() async {
+    if (_confirmConsentMessageShown != null) {
+      await _confirmConsentMessageShown();
+    }
+  }
+
   final VoidCallback? onSetupAnalytics;
 
+  final String? consentMessage;
+
```

Comment:
I made this non-nullable because I didn't want to assume that I can pass empty strings to the constructors within the test files.

Would it be okay if I made this non-nullable and for all tests, I can provide an empty string?

```dart
AnalyticsController _controller =
    AnalyticsController(enabled: false, firstRun: false, consentMessage: '');
```

Reply:
fine to make required. you can just set some bogus value for tests "Test consent message."